api/src/chat/repositories/block.repository.ts

```diff
@@ -89,16 +89,112 @@ export class BlockRepository extends BaseRepository<
       Block,
       'findOneAndUpdate'
     >,
-    _criteria: TFilterQuery<Block>,
+    criteria: TFilterQuery<Block>,
     _updates:
       | UpdateWithAggregationPipeline
       | UpdateQuery<Document<Block, any, any>>,
   ): Promise<void> {
     const updates: BlockUpdateDto = _updates?.['$set'];
+    if (updates?.category) {
```

Comment:
```suggestion
    if (updates?.category && criteria._id) {
```
Else we need to throw an error.